pkg/controller/che/che_controller.go

```diff
@@ -265,6 +265,14 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 	}
 
+	// Handle Che TLS certificates on Kubernetes like infrastructures
+	if instance.Spec.Server.TlsSupport && instance.Spec.Server.SelfSignedCert && !isOpenShift {
+		shouldReturn, reconsileResult, err := HandleCheTLSSecrets(instance, r)
```

Comment:
Can we return something like status structure? 
Something like this
https://github.com/eclipse/che-operator/blob/master/pkg/deploy/deployment.go#L42

Reply:
Theoretically we can, but I am against it. We need to return 3 values from the method and we can do it using embedded in the golang mechanism. Why do we need to introduce new abstraction then?

Reply:
+1